client/src/components/admin/AdminPhotos.vue

```diff
@@ -0,0 +1,58 @@
+<script setup lang="ts">
+import { getbaseURL } from "@/helper/baseUrlHelper";
+import ApiService from "@/services/ApiService";
+import { computed, ref } from "vue";
+import { useRouter } from "vue-router";
+
+const router = useRouter();
+
+const year = ref(new Date().getFullYear());
+const showSpinner = ref(false);
+const archivPath = ref("");
+
+const onCreate = async () => {
+  try {
+    showSpinner.value = true;
+    const response = await ApiService.createPhotoArchivOfYear(year.value);
+    archivPath.value = response.data;
+  } catch (error) {
+    console.log(error);
+    router.push({
+      name: "NetworkError",
+    });
+  } finally {
+    showSpinner.value = false;
+  }
+};
+
+const archivDownloadUrl = computed(() => {
+  const baseURL = getbaseURL();
+  const escapedPath = archivPath.value.replaceAll("/", "%2F");
+  return baseURL + "flights/photos/download/" + escapedPath;
+});
+</script>
+
+<template>
+  <div id="adminPhotoDownloadPanel" class="py-3">
+    <button
+      class="col-2 btn btn-outline-primary btn-sm bi bi-caret-right me-3"
+      :disabled="showSpinner || archivPath.length > 0"
+      @click="onCreate"
+    >
+      Erstelle Archiv
+      <div
+        v-if="showSpinner"
+        class="spinner-border spinner-border-sm"
+        role="status"
+      >
+        <span class="visually-hidden">Loading...</span>
+      </div>
+    </button>
+    <BaseInput v-model="year" class="mt-3" label="Jahr" />
```

Comment:
A dropdown would be better UX.

Reply:
Also, button remains uncklickable if the year is changed after a archive is created and it is possible to create a download link for the year 2023 that points to an invalid .zip

Reply:
Nur für den Adminbereich... s.o.
War mir zuviel Aufwand das alles einzubauen für eine Sache die nur einmal im Jahr benutzt wird.